netbox/page_test.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2017 The go-netbox Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netbox
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+type testOptions struct{}
+
+func (t *testOptions) values() (url.Values, error) {
+	return url.Values{
+		"Hello": []string{"World"},
+		"limit": []string{"30"},
+	}, nil
+}
+
+func TestPageValues(t *testing.T) {
+
+	var tests = []struct {
+		desc string
+		page *Page
+		want url.Values
+		err  error
+	}{
+		{
+			desc: "nil page",
+			page: nil,
+			want: nil,
+			err:  errors.New("page not defined"),
+		},
+		{
+			desc: "options nil",
+			page: NewPage(nil, "/", nil),
+			want: url.Values{
+				"limit":  []string{"50"},
+				"offset": []string{"0"},
+			},
+			err: nil,
+		},
+		{
+			desc: "options set",
+			page: NewPage(nil, "/", &testOptions{}),
+			want: url.Values{
+				"Hello":  []string{"World"},
+				"limit":  []string{"50"},
+				"offset": []string{"0"},
+			},
+			err: nil,
+		},
+	}
+
+	for i, tt := range tests {
+		res, err := tt.page.values()
+		if want, got := tt.err, err; !reflect.DeepEqual(want, got) {
+			t.Fatalf("[%d] %s - unecpected error:\n- want: %v\n-  got: %v", i, tt.desc, want, got)
```

Comment:
Now that subtests are available, we can use them instead of mentioning the test index and name in each error.

```go
for _, tt := range tests {
    t.Run(tt.desc, func(t *testing.T) {
        // Write tests as normal
    })
}
```